app/models/race.rb

```diff
@@ -33,14 +33,14 @@ def self.unfinished
   end
 
   def self.finished
-    joins(:entries).where.not(entries: {finished_at: nil, forfeited_at: nil}).distinct
+    where.not(id: unfinished)
```

Comment:
Would it be more efficient to do `with_ends.where(ended_at: nil)`?

Reply:
With the way `with_ends` is setup, we can't use it in a where clause.  I tried to get it to work, and was able to by using a from clause but as soon as you chained another scope it just fell apart.  Since this was the only thing I had in mind to use `ended_at` in a where clause, I just decided to go with this basic way of checking instead.

Reply:
Ohh ok gotcha so this is where you were having trouble with that

Reply:
If you want to tinker around with it, something as simple as `Race.select('*').from(Race.with_ends).where('ended_at IS NOT NULL')` was what kind of worked.  I'm not sure if it was because of the other calls to `with_end` or just the other scopes in general that were messing things up.